### PR TITLE
ddl: check DDL history job in test

### LINKED ISSUE #33134: Some DDL queries are written incorrectly
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

I found that some DDL's queries are written incorrectly from this PR #33079
For example, 
`unlock tables`, which is called after session close.
`cache table`, whose query is covered by some internal DML.

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: ref #33078, and fix #33134

Problem Summary:
Do some checks for complete/rollback DDL job.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by crazycs520 — APPROVED** [review]
LGTM

File: ddl/ddl.go [hunk 1]
```diff
@@ -610,13 +611,62 @@ func recordLastDDLInfo(ctx sessionctx.Context, job *model.Job) {
 	ctx.GetSessionVars().LastDDLInfo.SeqNum = job.SeqNum
 }
 
+func checkHistoryJobInTest(ctx sessionctx.Context, historyJob *model.Job) {
```
Comment (zimulala) [inline ddl/ddl.go:614]: Could we put it to ddl_test.go?

Comment (wjhuang2016) [reply ddl/ddl.go:614]: It's called during the normal logic.

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**